mlir/lib/Tools/PDLL/Parser/Parser.cpp

```diff
@@ -3048,6 +3166,38 @@ Parser::createTupleExpr(SMRange loc, ArrayRef<ast::Expr *> elements,
   return ast::TupleExpr::create(ctx, loc, elements, elementNames);
 }
 
+FailureOr<ast::DeclRefExpr *>
+Parser::createNativeCall(SMRange loc, StringRef nativeFuncName,
+                         MutableArrayRef<ast::Expr *> arguments) {
+
+  FailureOr<ast::Expr *> nativeFuncExpr = parseDeclRefExpr(nativeFuncName, loc);
+  if (failed(nativeFuncExpr))
+    return emitError(nativeFuncName + " not found.");
+
+  if (!(*nativeFuncExpr)->getType().isa<ast::RewriteType>())
+    return emitError(nativeFuncName + " should be defined as a rewriter.");
+
+  FailureOr<ast::CallExpr *> nativeCall =
+      createCallExpr(loc, *nativeFuncExpr, arguments);
+  if (failed(nativeCall))
+    return failure();
+
+  // Create a unique anonymous name to use, as the name for this decl is not
+  // important.
+  std::string anonName =
+      llvm::formatv("{0}_{1}", nativeFuncName, anonymousDeclNameCounter++)
+          .str();
+  FailureOr<ast::VariableDecl *> varDecl = defineVariableDecl(
+      anonName, loc, (*nativeCall)->getType(), *nativeCall, {});
+  if (failed(varDecl))
+    return failure();
+
+  FailureOr<ast::DeclRefExpr *> arrayAttrReference =
+      createDeclRefExpr(loc, *varDecl);
+
+  return *arrayAttrReference;
```

Comment:
As you don't check the error before it should just be `return arrayAttrReference;`, I think